test/test-helper.el

```diff
@@ -123,3 +123,9 @@ list of substrings of `STR' each followed by its face."
     (when buffer-file-name
       (setq-local compilation-ask-about-save nil)))
 (add-hook 'rustic-mode-hook 'rustic-mode-auto-save-hook)
+
+(defun rustic-test--wait-till-finished (buffer)
```

Comment:
And btw, please don't put too much effort into making existing tests look better, I think it's not worth the time.

Reply:
> Maybe we should pass the process to the function since we already got it ?

We don't get in some places. Eg:

- https://github.com/psibi/rustic/blob/1b8e54d76e1364e2dfa9876eb06fdb604281434d/test/rustic-window-test.el#L10
- https://github.com/psibi/rustic/blob/1b8e54d76e1364e2dfa9876eb06fdb604281434d/test/rustic-window-test.el#L25

> And btw, please don't put too much effort into making existing tests look better, I think it's not worth the time.

This whole PR started out as fixing one of the flaky test and I just went ahead with fixing other flaky tests. :-) I re-ran the CI multiple times and it seems to have become stable.

Reply:
Ok nevermind.

Reply:
Thank you!